javascript/script.js

```diff
@@ -0,0 +1,185 @@
+// Assignment Code
+var generateBtn = document.querySelector("#generate");
+
+function generatePassword() {
+  // PLEASE ADD ALL YOUR CODE HERE
+
+  //                       STEP 1 :
+  // prompt bar asking the user to input a character length
+  // Also convert that number string in to a number type data
+
+  const userPasswordLength = function () {
+    // declare an is valid booleon
+
+    const characterLength = prompt(
+      "Enter the character length the of your password"
+    );
+
+    //                       STEP 2 :
+    // if statement that validates the user input matches the password length criterial
+
+    if (isNaN(characterLength)) {
+      alert("Please type a number");
+      return;
+    } else {
+      const inputNumValue = Number(characterLength); // the input string is converted to a number
+      if (inputNumValue < 8 || inputNumValue > 128) {
+        alert(
+          "Password length is not valid. Pick a length between 8 and 128 and press cancel"
+        );
+        return;
+      } else {
+        return inputNumValue;
+      }
+    }
+  };
+
+  //                       STEP 3 :
+  //  function for confirm bar for whether the user wants lowercase characters in password
+
+  function lowerCaseOption() {
+    const option = confirm(
+      "Press OK if you want your password to have lowercase characters"
+    );
+    return option;
+  }
+
+  //                       STEP 4 :
+  //  function for confirm bar for whether the user wants uppercase characters in password
+
+  function upperCaseOption() {
+    const option = confirm(
+      "Press OK if you want your password to have uppercase characters"
+    );
+    return option;
+  }
+
+  //                       STEP 5 :
+  //  function for confirm bar for whether the user wants numbers characters in password
+
+  function numericOption() {
+    const option = confirm(
+      "Press OK if you want your password to have Numbers"
+    );
+    return option;
+  }
+
+  //                       STEP 6 :
+  //  function for confirm bar for whether the user wants special characters in password
+  function specialCharacterOption() {
+    const option = confirm(
+      "Press OK if you want your password to have special characters"
+    );
+    return option;
+  }
+
+  //                       STEP 7 :
+  //  in a function declare a list of lowercase , uppercase , numbers and special characters in a string
+  // convert the string into an array and then generate a random
+
+  function randomLowerCaseCharacter() {
```

Comment:
You have quite a lot of repeated code over these 4 functions. Is there any way we can make this function more reusable, so that we just have it once to be used with all 4 character types? As your code gets longer, you will want to avoid repeated code as much as possible - let me know if you're not sure how we could do this.